week1/citibike.sh

```diff
@@ -4,20 +4,29 @@
 #
 
 # count the number of unique stations
-
+    cut -d, -f4 201402-citibike-tripdata.csv | sort | uniq | wc -l
+  
 # count the number of unique bikes
-
+    cut -d, -f12 201402-citibike-tripdata.csv | sort | uniq | wc -l
+  
 # count the number of trips per day
-
+    cut -d, -f2 201402-citibike-tripdata.csv | cut -c2-11 | sort -n | uniq -c
+awk -F\" 'NR!=1 {sum+=$2} END {print sum/NR}' 201402-citibike-tripdata.csv
 # find the day with the most rides
+    cut -d, -f2 201402-citibike-tripdata.csv | cut -d ' ' -f1 | sort | uniq -c | sort -nr | head -n1
 
 # find the day with the fewest rides
+    cut -d, -f2 201402-citibike-tripdata.csv | cut -d ' ' -f1 | sort | uniq -c | sort | head -n1
 
 # find the id of the bike with the most rides
+    cut -d, -f12 201402-citibike-tripdata.csv | sort | uniq -c | sort -nr | head -n1
 
 # count the number of rides by gender and birth year
+    cut -d, -f15,14 201402-citibike-tripdata.csv | sort | uniq -c | grep -v "birth"
 
 # count the number of trips that start on cross streets that both contain numbers (e.g., "1 Ave & E 15 St", "E 39 St & 2 Ave", ...)
-
+    cut -d, -f5 201402-citibike-tripdata.csv | grep '[0-9].*&.*[0-9]' | wc -l
 
 # compute the average trip duration
+    awk -F\" 'NR!=1 {sum+=$2} END {print sum/NR}' 201402-citibike-tripdata.csv
```

Comment:
Great! And you use the NR != 1 trick for killing headers!

Reply:
That was bothering me even if it did not matter.